FullGroupJoin test improvements

fsateler opened the pull request:
Fixes #325

atifaziz commented on MoreLinq.Test/FullGroupJoinTest.cs:
```diff
@@ -29,7 +31,11 @@ public void FullGroupIsLazy()
             var listB = new BreakingSequence<int>();
 
-            listA.FullGroupJoin(listB, x => x, x => x, DummySelector);
+            int BreakingSelector(int _, IEnumerable<int> a, IEnumerable<int> b)
```
We already have `BreakingFunc` so best to just re-use it as follows:

```c#
listA.FullGroupJoin(listB, x => x, x => x, BreakingFunc.Of<int, IEnumerable<int>, IEnumerable<int>, int>());
```

fsateler commented:
@atifaziz thanks for the review. I've changed the PR accordingly.

atifaziz commented on MoreLinq.Test/FullGroupJoinTest.cs:
```diff
@@ -125,6 +126,18 @@ public void FullGroupPreservesOrder()
+        {
+            switch (overloadCase) {
+            case OverloadCase.ResultSelector:
+                return listA.FullGroupJoin(listB, getKey, getKey, (k, f, s) => (k, f, s));
```
Do you prefer spelling it out as `(k, f, s) => (k, f, s)` instead of just using `TupleValue.Create` as a method group?

atifaziz commented on MoreLinq.Test/FullGroupJoinTest.cs:
```diff
@@ -102,29 +108,39 @@ public void FullGroupPreservesOrder()
-            foreach (var res in result) {
-                res.first.AssertSequenceEqual(listA.Where(t => t.Item1 == res.key).ToArray());
-                res.second.AssertSequenceEqual(listB.Where(t => t.Item1 == res.key).ToArray());
+            foreach (var res in result)
```
Non-essential but perhaps this can be simplified to?

```c#
            foreach (var (key, fst, snd) in result)
            {
                fst.AssertSequenceEqual(listA.Where(t => t.Item1 == key).ToArray());
                snd.AssertSequenceEqual(listB.Where(t => t.Item1 == key).ToArray());
            }
```

atifaziz commented on MoreLinq.Test/FullGroupJoinTest.cs:
```diff
@@ -102,29 +108,39 @@ public void FullGroupPreservesOrder()
+            switch (overloadCase)
+            {
+                case OverloadCase.ResultSelector:
+                    return listA.FullGroupJoin(listB, getKey, getKey, (k, f, s) => (k, f, s));
```
Do you prefer spelling out `(k, f, s) => (k, f, s)` over just using the `ValueTuple.Create` as method group?

fsateler commented:
> Do you prefer spelling out (k, f, s) => (k, f, s) over just using the ValueTuple.Create as method group?

Not really. Changed

I also changed to unpack the tuple in the loop.

atifaziz commented:
@fsateler What do you make of 9fa49f6e386320e16bd129a0f8c68e4b60701333? Happy to revert but just went easier on my aging & sore eyes. 😄 Usually I don't do statically import enums but here it's really scoped to a specific case, one test file and makes for easier reading.

fsateler commented:
@atifaziz Looks good to me. I had never done that, but I agree that for
this very local use it looks good.

On Thu, Jul 6, 2017 at 12:04 PM, Atif Aziz <notifications@github.com> wrote:

> @fsateler <https://github.com/fsateler> What do you make of 9fa49f6
> <https://github.com/morelinq/MoreLINQ/commit/9fa49f6e386320e16bd129a0f8c68e4b60701333>?
> Happy to revert but just went easier on my aging & sore eyes. 😄 Usually
> I don't do statically import enums but here it's really scoped to a
> specific case, one test file and makes for easier reading.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/morelinq/MoreLINQ/pull/330#issuecomment-313441634>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQsHeXA71nU_ZJR5XAyt3C1U7i7_kB2ks5sLQWZgaJpZM4ONtw7>
> .
>



-- 

Saludos,
Felipe Sateler
